main.tf

```diff
@@ -1 +1,188 @@
+}
+
+# By calling this data source we are accessing the storage service
+# account and therefore, Google will created for us. If we don't
+# ask for it, Google doesn't create it by default, more docs at:
+# => https://cloud.google.com/storage/docs/projects#service-accounts
+#
+# If the service account is not there, we could add a local-exec
+# provisioner to call an API that is documented at:
+# => https://cloud.google.com/storage-transfer/docs/reference/rest/v1/googleServiceAccounts/get
+data "google_storage_project_service_account" "lw" {
+  project = local.project_id
+}
+
+resource "google_pubsub_topic_iam_binding" "topic_publisher" {
+  members = local.logging_sink_writer_identity
+  role    = "roles/pubsub.publisher"
+  project = local.project_id
+  topic   = google_pubsub_topic.lacework_topic.name
+  depends_on = [google_pubsub_topic.lacework_topic]
+}
+
+resource "google_pubsub_subscription" "lacework_subscription" {
+  project                    = local.project_id
+  name                       = "${var.prefix}-${local.project_id}-lacework-subscription-${random_id.uniq.hex}"
+  topic                      = google_pubsub_topic.lacework_topic.name
+  ack_deadline_seconds       = 300
+  message_retention_duration = "432000s"
+  labels                     = merge(var.labels, var.pubsub_subscription_labels)
+  depends_on = [google_pubsub_topic.lacework_topic]
+}
+
+resource "google_logging_project_sink" "lacework_project_sink" {
+  count                  = length(var.existing_sink_name) > 0 ? 0 : (local.org_integration ? 0 : 1)
+  project                = local.project_id
+  name                   = local.sink_name
+  destination            = "pubsub.googleapis.com/${google_pubsub_topic.lacework_topic.id}"
+  unique_writer_identity = true
+
+  filter = local.log_filter
+  depends_on = [google_pubsub_topic.lacework_topic]
+}
+
+resource "google_logging_organization_sink" "lacework_organization_sink" {
+  count            = length(var.existing_sink_name) > 0 ? 0 : ((local.org_integration) ? 1 : 0)
+  name             = local.sink_name
+  org_id           = var.organization_id
+  destination      = "pubsub.googleapis.com/${google_pubsub_topic.lacework_topic.id}"
+  include_children = true
+
+  filter = local.log_filter
+  depends_on = [google_pubsub_topic.lacework_topic]
+}
+
+resource "google_pubsub_subscription_iam_binding" "lacework" {
+  project      = local.project_id
+  role         = "roles/pubsub.subscriber"
+  members      = ["serviceAccount:${local.service_account_json_key.client_email}"]
+  subscription = google_pubsub_subscription.lacework_subscription.name
+  depends_on = [google_pubsub_subscription.lacework_subscription]
+}
+
+resource "google_project_iam_member" "for_lacework_service_account" {
+  project = local.project_id
+  role    = "roles/storage.objectViewer"
+  member  = "serviceAccount:${local.service_account_json_key.client_email}"
+}
+
+resource "google_organization_iam_member" "for_lacework_service_account" {
```

Comment:
hmmm this probably isn't required. If it is an org level integration we require the user to pass the org id anyway, so we should already have this info. I'll remove this.

Reply:
So I've since remembered that we actually do some validation that we can access the project/org depending on the integration type, when creating the integration. I believer we may need to have the browser role to ensure that we can perform this basic validation on integration creation.

Reply:
@mikelaramie I actually have a ticket to re-work the error messages in the api-server tier to be more descriptive. 
So this might be a good time to review the integration validation we have in place. Perhaps verifying that we can get the org display name and project details isn't really required and we just need to verify that the pub/sub subscription exists and we can access the subscription with the credentials?

Reply:
Making this change should allow us to remove the org and project iam member resources. https://github.com/lacework/rainbow/pull/9166